### ISSUE TITLE
Should libsass have an image-url function?

### ISSUE THREAD

**Issue by michaek** [sass/libsass#489]
See https://github.com/sass/libsass/blob/fca1f75a14fe5336c7b1a4b38c69bda7fece714f/functions.cpp#L1179

Because image-url is a Compass function, not a Sass function, it seems odd for it to be included in libsass. Building in any URL functions requires "Sass" to know about your paths - Compass has embraced that pattern, but I don't know if we should assume that Sass does.

**Comment by xzyfer** [comment]
Unless there are any objections I'm going to make `image-url` produce a deprecation warning in 3.1.1. Then let's figure out what we are going to do about it for 3.2.

**Comment by mgreter** [comment]
I'm all in for this! Maybe @hcatlin can pull some strings to get this finally done!? These features should be possible to implement via custom functions now IMO!? I can offer my help to give advice on how to do it via custom functions! But we'd need someone who actually uses this feature to work with us!

**Comment by hcatlin** [comment]
Hey @chriseppstein, has the compass team gotten to a pure-Sass image-url() implementation yet?

**Comment by chriseppstein** [comment]
No sass-based definition of image-url in compass since compass requires ruby, there's no point. Any given project can probably implement a trivial image-url() helper for that project.

**Comment by hcatlin** [comment]
From @jonsuh and @roytomeij:

````
$image-url-path: “../images/” !default;
@function image-url($url) {
  @return url("#{$image-url-path}#{$url}");
}
````

**Comment by hcatlin** [comment]
@chriseppstein Sorry, I thought there was an effort to port some functions from Ruby to Sass, and didn't know if this was one of them.
@xzyfer I think I've agreed since the top to remove it...

**Comment by chriseppstein** [comment]
@hcatlin There is. This will be one of them :)

I wouldn't say this though:

> `image_path` has been removed from libsass till it becomes part of the Sass spec.

Because it's not something that Sass itself would ever add. Maybe this:

> `image_path` has been removed from libsass because it's not part of the Sass spec.

**Comment by xzyfer** [comment]
Ok cool. I'm going to remove the guts of this and add a deprecation message. 3.2. Is pretty close atm so this is a good a time as any.

**Comment by adyballa** [comment]
@hcatlin
btw
I have good use of image_path, because I dont know the prefix in sass. So this function will not work

**Comment by callumacrae** [comment]
I'm confused. Was this deprecated or removed?

Updating to node-sass 3.1 broke all the images on our website. This is from the output CSS:

```css
.sun-rays {
  background-image: image-url("bg-with-sunrays-large.png");
  background-size: cover;
  background-repeat: no-repeat; }
```

I'm not getting an error using node-sass through gulp-sass. It's just outputting image-url and silently breaking.

Could I suggest that breaking changes not be hidden at the bottom of the release notes? Also, could you collaborate with the node-sass folks so that breaking changes in libsass are mentioned there, too? There was nothing in the node-sass 3.1 release notes about this function being removed.

Also, in the future, a soft deprecation a version ahead of the head deprecation would be massively appreciated! I had no idea this feature was non-standard until it disappeared.

**Comment by xzyfer** [comment]
Hi @callumacrae, we work very closely with the node-sass team. I believe this change was clearly communicated, and there were several beta releases.

![screen shot 2015-05-18 at 8 56 25 pm](https://cloud.githubusercontent.com/assets/579928/7679475/5f2678aa-fda0-11e4-8a1a-91f9b6105533.png)

As it stands Libsass does not follow semver, and will not until 3.4. We did soft deprecate this feature in 3.2.0 and subsequently removed in a later patch since the deprecation was causing more troubles to users.

**Comment by xzyfer** [comment]
Lets consolidate any remaining discussion in https://github.com/sass/node-sass/issues/965.

**Comment by danielrz** [comment]
what about the $cache-buster parameter? does LibSass supports cache busting with images?

**Comment by xzyfer** [comment]
No it does not.

**Comment by swarajban** [comment]
@callumacrae we're having the same problem after upgrading. What did you do to fix?

**Comment by xzyfer** [comment]
You need to implement your own `image-url` function. It's very straight forward to do.

**Comment by idflood** [comment]
@swarajban here is a simple implementation:

    sass({
      functions: {
        'image-url($img)': function(img) {
          return new sass.types.String('url("images_directory/' + img.getValue() + '")');
        }
      }
    })

**Comment by swarajban** [comment]
Thanks @xzyfer and @idflood . I ended up using asset functions from [this node package](https://github.com/fetch/node-sass-asset-functions)

**Comment by GuyPaddock** [comment]
Also see: https://github.com/fetch/node-sass-asset-functions
